FluxC - Media Integration - bind/unbind store in the EditPost should be re thinked

We should reuse `doUnbindUploadService` from MediaBrowserActivity to avoid any unbinding issues

```
java.lang.RuntimeException: Unable to stop activity {org.wordpress.android.beta/org.wordpress.android.ui.posts.EditPostActivity}: java.lang.IllegalArgumentException: Service not registered: org.wordpress.android.ui.posts.EditPostActivity$10@39b8d19a
    at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:3344)
    at android.app.ActivityThread.handleStopActivity(ActivityThread.java:3390)
    at android.app.ActivityThread.access$1100(ActivityThread.java:144)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1307)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:135)
    at android.app.ActivityThread.main(ActivityThread.java:5221)
    at java.lang.reflect.Method.invoke(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
 Caused by: java.lang.IllegalArgumentException: Service not registered: org.wordpress.android.ui.posts.EditPostActivity$10@39b8d19a
    at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1024)
    at android.app.ContextImpl.unbindService(ContextImpl.java:1810)
    at android.content.ContextWrapper.unbindService(ContextWrapper.java:550)
    at org.wordpress.android.ui.posts.EditPostActivity.onStop(EditPostActivity.java:443)
    at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1261)
    at android.app.Activity.performStop(Activity.java:6085)
    at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:3341)
    at android.app.ActivityThread.handleStopActivity(ActivityThread.java:3390) 
    at android.app.ActivityThread.access$1100(ActivityThread.java:144) 
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1307) 
    at android.os.Handler.dispatchMessage(Handler.java:102) 
    at android.os.Looper.loop(Looper.java:135) 
    at android.app.ActivityThread.main(ActivityThread.java:5221) 
    at java.lang.reflect.Method.invoke(Native Method) 
    at java.lang.reflect.Method.invoke(Method.java:372) 
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899) 
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694) 
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java
@@ -223,6 +223,10 @@ public class EditPostActivity extends AppCompatActivity implements EditorFragmen
     @Inject PostStore mPostStore;
     @Inject MediaStore mMediaStore;
 
+    // Upload service
+    private MediaUploadService.MediaUploadBinder mUploadService;
+    private boolean mUploadServiceBound;
+
     private SiteModel mSite;
 
     // for keeping the media uri while asking for permissions
@@ -439,9 +443,6 @@ public class EditPostActivity extends AppCompatActivity implements EditorFragmen
 
     @Override
     public void onStop() {
-        if (mUploadService != null) {
-            unbindService(mUploadConnection);
-        }
         EventBus.getDefault().unregister(this);
         super.onStop();
     }
@@ -464,6 +465,7 @@ public class EditPostActivity extends AppCompatActivity implements EditorFragmen
     protected void onDestroy() {
         AnalyticsTracker.track(AnalyticsTracker.Stat.EDITOR_CLOSED);
         mDispatcher.unregister(this);
+        doUnbindUploadService();
         super.onDestroy();
     }
 
@@ -823,31 +825,6 @@ public class EditPostActivity extends AppCompatActivity implements EditorFragmen
     @Override
     public void onUploadProgress(MediaModel media, float progress) {
         mEditorMediaUploadListener.onMediaUploadProgress(String.valueOf(media.getId()), progress);
-    }
-
-    private void startMediaUploadService(ArrayList<MediaModel> mediaToUpload) {
-        if (!NetworkUtils.isNetworkAvailable(this)) {
-            AppLog.v(AppLog.T.MEDIA, "Unable to start MediaUploadService, internet connection required.");
-            return;
-        }
-
-        if (mUploadService != null) {
-            if (mediaToUpload != null && !mediaToUpload.isEmpty()) {
-                for (MediaModel media : mediaToUpload) {
-                    mUploadService.addMediaToQueue(media);
-                }
-                mediaToUpload.clear();
-            }
-        } else if (NetworkUtils.isNetworkAvailable(this)) {
-            Intent intent = new Intent(this, MediaUploadService.class);
-            intent.putExtra(MediaUploadService.SITE_KEY, mSite);
-            if (mediaToUpload != null) {
-                intent.putExtra(MediaUploadService.MEDIA_LIST_KEY, mediaToUpload);
-                bindService(intent, mUploadConnection, Context.BIND_AUTO_CREATE | Context.BIND_ABOVE_CLIENT);
-                mediaToUpload.clear();
-            }
-            startService(intent);
-        }
     }
 
     private void launchPictureLibrary() {
@@ -1856,8 +1833,6 @@ public class EditPostActivity extends AppCompatActivity implements EditorFragmen
         }
     }
 
-    private MediaUploadService.MediaUploadBinder mUploadService;
-
     private ServiceConnection mUploadConnection = new ServiceConnection() {
         @Override
         public void onServiceConnected(ComponentName name, IBinder service) {
@@ -1876,6 +1851,18 @@ public class EditPostActivity extends AppCompatActivity implements EditorFragmen
         }
     };
 
+
+    private void doBindUploadService(Intent intent) {
+        bindService(intent, mUploadConnection, Context.BIND_AUTO_CREATE | Context.BIND_ABOVE_CLIENT);
+        mUploadServiceBound = true;
+    }
+
+    private void doUnbindUploadService() {
+        if (mUploadServiceBound) {
+            unbindService(mUploadConnection);
+        }
+    }
+
     /**
      * Starts the upload service to upload selected media.
      */
@@ -1883,7 +1870,7 @@ public class EditPostActivity extends AppCompatActivity implements EditorFragmen
         if (mUploadService == null) {
             Intent intent = new Intent(this, MediaUploadService.class);
             intent.putExtra(MediaUploadService.SITE_KEY, mSite);
-            bindService(intent, mUploadConnection, Context.BIND_AUTO_CREATE | Context.BIND_ABOVE_CLIENT);
+            doBindUploadService(intent);
             startService(intent);
         } else if (mPendingUploads != null && !mPendingUploads.isEmpty()) {
             for (MediaModel media : mPendingUploads) {
@@ -2008,6 +1995,7 @@ public class EditPostActivity extends AppCompatActivity implements EditorFragmen
         if (mUploadService == null) {
             startMediaUploadService();
         } else {
+            // TODO: FluxC integration on retry?
         }
         AnalyticsTracker.track(Stat.EDITOR_UPLOAD_MEDIA_RETRIED);
     }

